src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -103,7 +109,7 @@ def listRecords(self, metadataPrefix=None, set=None, cursor=None, from_=None,
         return data
 
     def getRecord(self, metadataPrefix, identifier):
-        record = CatalogRecord.objects.filter(
+        record = CatalogRecord.objects.filter(data_catalog_id__in=self._get_set_filter(),
             **{ 'research_dataset__contains': {'urn_identifier': identifier } }).first()
```

Comment:
Leaving a note as per the discussion:

Since only one result is intended to be retrieved (by using .first()) here, using .get() is more well suited for the purpose compared to .filter(). .get() will raise an error if there are more than one results found (exception name: CatalogRecord.MultipleObjectsReturned (although in this case this error would of course be highly unlikely)), or no records found at all (exception name: CatalogRecord.DoesNotExist), so a try: except pattern would replace the current one.

For future reference, .filter() would be ok to use if intending to retrieve only a single field i.e. .filter().values_list('id', Flat=True).first(), since that can not be done with .get().

Additionally, while this works **{ 'research_dataset__contains': {'urn_identifier': identifier } }, it could be reduced to just research_dataset__contains={'urn_identifier': identifier }